sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ProjectionOverSchema.scala

```diff
@@ -49,6 +49,10 @@ case class ProjectionOverSchema(schema: StructType) {
               s"unmatched child schema for GetArrayStructFields: ${projSchema.toString}"
             )
         }
+      case MapKeys(child) =>
+        getProjection(child).map { projection => MapKeys(projection) }
+      case MapValues(child) =>
+        getProjection(child).map { projection => MapValues(projection) }
```

Comment:
All the extraction expressions listed in `SelectedField` basically should appear in this file?

Reply:
The expressions which can be part of an extractor chain, should appear in `ProjectionOverSchema`.